pkg/network/common/dns_test.go

```diff
@@ -160,8 +160,9 @@ func TestAddDNS(t *testing.T) {
 			if !ipsEqual(d.ips, test.ips) {
 				t.Fatalf("Test case: %s failed, expected IPs: %v, got: %v for the domain %q", test.testCase, test.ips, d.ips, test.domainName)
 			}
-			if d.ttl.Seconds() != test.ttl {
-				t.Fatalf("Test case: %s failed, expected TTL: %g, got: %g for the domain %q", test.testCase, test.ttl, d.ttl.Seconds(), test.domainName)
+			normalizedTTL := normalizeTTL(time.Duration(test.ttl) * time.Second)
```

Comment:
If you made `test.ttl` a `time.Duration` you could get rid of a lot of casting/multiplying